Add provisional product specification for burnt area rasters.

james-blitzm opened the pull request.

james-blitzm committed:
Add provisional product specification for burnt area rasters.

james-blitzm committed:
Change product family to "ba"

james-blitzm committed:
Fix lining issue.

james-blitzm committed:
Rename product file.

james-blitzm committed:
More linter fixes

james-blitzm commented:
@jeremyh @uchchwhash It looks like the above CI failures already exist in the project and aren't related to my changes here?

uchchwhash commented:
@james-blitzm that is correct!

uchchwhash reviewed (approved):
LGTM!

jeremyh reviewed (commented):
Thanks James. I've been meaning to add [eo3-validate](https://github.com/GeoscienceAustralia/eo-datasets#validator) to our CI checks -- it's reporting a schema error:
```
❯ eo3-validate 'https://raw.githubusercontent.com/GeoscienceAustralia/digitalearthau/4d631b8e1524812417b030ac2a15eb94fe9b8105/digitalearthau/config/eo3/products-aws/ga_s2_ba_provisional_3.yaml'
0 ODC products
✗ https://raw.githubusercontent.com/GeoscienceAustralia/digitalearthau/4d631b8e1524812417b030ac2a15eb94fe9b8105/digitalearthau/config/eo3/products-aws/ga_s2_ba_provisional_3.yaml
	W missing_suffix Document looks like a product but does not have filename extension ".odc-product.yaml"
	E document_schema (measurements.3.units) 1 is not of type 'string' 

failure: 1 warning, 1 error
```

Ideally we want to run the validator with a dataset too to make sure they match each-other -- is there an example dataset somewhere?

jeremyh commented:
The CI was fixed in the main branch, so will be accurate now.

james-blitzm committed:
Rebase from develop and fix production validation error.

james-blitzm committed:
Fix filename

uchchwhash commented on digitalearthau/config/eo3/products-aws/ga_s2_ba_provisional_3..odc-product.yaml:
```diff
@@ -0,0 +1,34 @@
+  product:
+    name: ga_s2_ba_provisional_3
+  properties:
+    eo:platform: sentinel-2
```
this line will have to be removed now

james-blitzm replied:
I've removed this field.

james-blitzm committed:
Remove incorrect platform property.

james-blitzm committed:
Rename measurements to match alchemist process.

james-blitzm commented:
Some recent changes:
1. Renamed the file as it was wrong.
2. Removed the `platform` field
3. Updated the measurement names to match the generated outputs (from alchemist).

I have been able to successfully index burn rasters using this specification in my dev environment.

jeremyh commented:
Could you put one of your dataset documents into the `digitalearthau/config/eo3/products-aws/examples` folder? 

That way the linter will correctly check that bands +metadata line up with it, etc.

james-blitzm committed:
fix measurement names

james-blitzm committed:
Add example dataset medatadata file

james-blitzm commented:
> Could you put one of your output dataset documents into the `digitalearthau/config/eo3/products-aws/examples` folder?
> 
> That way the linter will correctly check that measurements+metadata line up with it, etc. And it's a useful reference to have.

Added in latest commit -  https://github.com/GeoscienceAustralia/digitalearthau/pull/304/commits/381b1ea636677415ea2e31f4870fdc448570e093
